build_for_android.sh

```diff
@@ -0,0 +1,159 @@
+  echo "## Advanced features and Environment variables that may be set..."
+  echo "NDK_STANDALONE_TOOLCHAIN - A standalone toolchain is required to build full capabilities."
+  echo "NDK_HOME                 - This will allow to use the ndk-build, and a full standalone toolchain is not required.  If this is set, the scanmem executable will be built with ndk-build, and not the standalone toolchain.  This needs to be used if the target Android version is below 5.0.  This requirement may be fixed in the future."
+  echo "ANDROID_NDK_HOME         - Same as NDK_HOME"
+  echo "HOST                     - Compiler architecture that will be used for cross-compiling, default is arm-linux-androideabi"
+  echo "SCANMEM_HOME             - Path which has scanmem sources, and will be used to build scanmem.  Default current directory"
+  echo "LIBREADLINE_DIR          - Path which has libreadline sources: Default is to download sources"
+  echo "NCURSES_DIR              - Path which has ncurses sources: Default is to download sources"
+  echo
+  exit 0
+fi
+
+# Build with Android.mk
+# Do from scanmem directory, or SCANMEM_HOME
+ANDROID_MK(){
+  # Readline support with ndk-build does not work
+#  if [ -f ./jni/libreadline.a ] && [ -f ./jni/libncurses.a ]; then
+#    export WITH_READLINE=1
+#    export HAVE_LIBREADLINE=1
+#  fi
+  local ndkhome
+  if [ ! "x${NDK_HOME}" = "x" ]; then
+    ndkhome="${NDK_HOME}"
+  elif [ ! "x${ANDROID_NDK_HOME}" = "x"]; then
+    ndkhome="${ANDROID_NDK_HOME}"
+  else
+    echo "Error: Please set \$NDK_HOME env variable." 1>&2
+    exit 1
+  fi
+
+  # TODO: Configuring without cross-compile may give bad values in config.h
+  if [ ! -f config.h ]; then
+    ./configure --without-readline
+  fi
+  "${ndkhome}/ndk-build"
+}
+
+# Resolve ndk toolchain or other
+if [ "x${NDK_STANDALONE_TOOLCHAIN}" = "x" ]; then
+  echo "NDK_STANDALONE_TOOLCHAIN was not found.  Please enter the toolchain path:"
+  read NDK_STANDALONE_TOOLCHAIN
+  # Nothing entered
+  if [ "x${NDK_STANDALONE_TOOLCHAIN}" = "x" ]; then
+    # NDK home, but no toolchain, build with Android make system
+    if [ ! "x${NDK_HOME}" = "x" ] || [ ! "x${ANDROID_NDK_HOME}" = "x" ]; then
+      ANDROID_MK
+      exit 0
+    else
+      echo "Error: Please set \$NDK_STANDALONE_TOOLCHAIN env variable." 1>&2
+      echo "To build with Android.mk, without building libreadline, please set the \$NDK_HOME env variable instead." 1>&2
+      exit 1
+    fi
+  fi
+fi
+export SYSROOT="${NDK_STANDALONE_TOOLCHAIN}/sysroot"
+export PATH="${NDK_STANDALONE_TOOLCHAIN}/bin:${PATH}"
+
+# Host architecture
+if [ "x${HOST}" = "x" ]; then
+  HOST=arm-linux-androideabi
+  echo "Env variable \$HOST, host architecture, is not specified.  Defaulting to ${HOST}"
+fi
+
+# Build and return directory
+if [ "x${SCANMEM_HOME}" = "x" ]; then
+  export SCANMEM_HOME="$(pwd)"
+else
+  cd "${SCANMEM_HOME}"
+fi
+
+# Processor count for make instructions
+procnum="$(cat /proc/cpuinfo | grep -e '^processor' | wc -l)"
```

Comment:
`cat /proc/cpuinfo` doesn't work on Mac OSX, maybe `cat /proc/cpuinfo 2>/dev/null` to hide the error message.